Added empty default for name parameter

maishsk opened the pull request:
##### SUMMARY
Fix for issue #38482


##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
ec2_ami

##### ANSIBLE VERSION
```
ansible 2.5.0
  config file = None
  configured module search path = [u'/Users/msaidelk/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/msaidelk/venv/lib/python2.7/site-packages/ansible
  executable location = /Users/msaidelk/venv/bin/ansible
  python version = 2.7.10 (default, Jul 15 2017, 17:16:57) [GCC 4.2.1 Compatible Apple LLVM 9.0.0 (clang-900.0.31)]
```


##### ADDITIONAL INFORMATION
See issue #38482
Submitted based instructions from #38514

Cherry picked the fix and backported...

Commit:
Added empty default
Fix for issue https://github.com/ansible/ansible/issues/38482

(cherry picked from commit 60a32771039c190a7e9f6e0eb4d0f6bf4c356fd4)

Commit:
removed additional check for name parameter

(cherry picked from commit 91357d07f4f9eb3714f582c1959ff54d25129500)

s-hertel commented on lib/ansible/modules/cloud/amazon/ec2_ami.py:
```diff
@@ -656,7 +656,7 @@ def main():
         root_device_name=dict(),
         delete_snapshot=dict(default=False, type='bool'),
-        name=dict(),
+        name=dict(default=''),
```
I would like to see this fixed differently on devel because now the user gets a traceback if they omit name and image id and the message does not indicate image_id a possibility for state=present. I've opened https://github.com/ansible/ansible/pull/38972 and if that is merged will cherry pick the commit onto this pull request.

s-hertel committed:
[ec2_ami] Ensure name or image_id is provided for state=present (#38972)

Add integration tests for backward compatibility and ensuring name or image_id is provided
(cherry picked from commit e2aa1155ba5889296d2f018fdf8284c33f5f4918)
